service/history/timerQueueProcessor.go

```diff
@@ -414,12 +414,14 @@ func newQueueProcessorRateLimiter(
 ) quotas.RateLimiter {
 	return quotas.NewMultiRateLimiter(
 		[]quotas.RateLimiter{
+			// consume from host rate limiter as it's usually the one throttles the traffic
+			// and avoid unnecessary consume and cancel on shard level rate limiter
+			hostRateLimiter,
```

Comment:
My understanding is if later ratelimiter (the shard level one) can't pass, previous reservations will be cancelled.
https://github.com/temporalio/temporal/blob/master/common/quotas/multi_rate_limiter_impl.go#L106

Reply:
Please test to verify.

Reply:
My local test shows it works fine. But theoretically one rogue shard can still affect others since other shard can try to get a host token before the rouge shard cancels it. So I reverted the change here to be safe.

The busy loop issue is still fixed.